sound/soc/intel/boards/bdw-rt5677.c

```diff
@@ -319,13 +324,15 @@ static struct snd_soc_dai_link bdw_rt5677_dais[] = {
 		SND_SOC_DAILINK_REG(fe, dummy, platform),
 	},
 
+#if !IS_ENABLED(CONFIG_SND_SOC_SOF_BROADWELL)
 	/* Non-DPCM links */
 	{
 		.name = "Codec DSP",
 		.stream_name = "Wake on Voice",
 		.ops = &bdw_rt5677_dsp_ops,
 		SND_SOC_DAILINK_REG(dsp),
 	},
+#endif
```

Comment:
this is wrong I believe. @cujomalainey please correct me if I am smoking.
the WakeOnVoice is handled by the rt5677 codec IIRC, so there's no reason why it should be commented out.
However we need to check that the FE override does not prevent this from working.

Reply:
@plbossart the problem is this is a non-DPCM link and we do not override it at all with SOF. This was one of the questions @dbaluta was asking as well.

Reply:
it's good if it's not overridden!

Reply:
@xiulipan care to explain why we need to comment this out?

Reply:
The only dependency on the platform is a DAPM link to force the I2S clk. If SOF can provide that clock then WoV should work fine with SOF as the audio bypasses the DSP on a SPI link.

Reply:
@cujomalainey can you elaborate here? I don't remember what the expectation is for clocks here, and quite frankly whether we use SOF or not the hardware is the same, so there's not really any magic we can do...

Reply:
The RT5677 needs the I2S MCLK running in order to drive its DSP. Locally we have 0d2135ecadb0b2eec5338a7587ba29724ddf612b reverted (the same patch that causes DMA boot issues on samus and DSP crashes on buddy.) By reverting it, the kernel allows the bdw DSP to suspend and leave the MCLK running so everything is fine for us.

If we don't have that clock bad things start to happen very quickly. All I2C writes silently fail is one example and the registers read back garbage values (usually 0xBEEF).

Reply:
The MCLK is not controlled by the DSP, did you mean SCLK or BCLK?

Reply:
Nope I mean MCLK. It is controlled by the DSP kernel driver, at least that is where it is in SST builds.

Reply:
@cujomalainey @plbossart I will test with this FE kept and see if we can find the link there.
Then I think we can decided if we need to keep this here.